### PR TITLE
Implement multik-cuda module

### PR DESCRIPTION
**Pull request by Idutvuker**
Created cuda module.
Implemented dot function for matrices and vectors.

### CODE REVIEW

File: multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    override fun <T : Number> inv(mat: MultiArray<T, D2>): NDArray<T, D2> {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T : Number, D : Dim2> solve(a: MultiArray<T, D2>, b: MultiArray<T, D>): NDArray<T, D> {
+        TODO("Not yet implemented")
+    }
+
+    private fun <T : Number, D: Dim2> isTransposedConsistent(x: MultiArray<T, D>): Boolean {
+        return x.transpose().consistent
```
Comment (devcrocod) [inline multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt:34]: Not the most optimal solution. You can do the same by comparing shape and strides. Make such a method in `NDArray` and call `isTranspose`.

File: multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    private fun <T : Number, D: Dim2> getConsistentOrTransposedConsistent(x: MultiArray<T, D>): Pair<MultiArray<T, D>, Boolean> {
+        return when {
```
Comment (devcrocod) [inline multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt:38]: You can change `return` to `=`

File: multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+    override fun <T : Number, D : Dim2> solve(a: MultiArray<T, D2>, b: MultiArray<T, D>): NDArray<T, D> {
+        TODO("Not yet implemented")
+    }
+
+    private fun <T : Number, D: Dim2> isTransposedConsistent(x: MultiArray<T, D>): Boolean {
+        return x.transpose().consistent
+    }
+
+    private fun <T : Number, D: Dim2> getConsistentOrTransposedConsistent(x: MultiArray<T, D>): Pair<MultiArray<T, D>, Boolean> {
```
Comment (devcrocod) [inline multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt:37]: Why return Pair?

File: multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        return when {
+            x.consistent -> x to false
+            x.dim.d == 2 && isTransposedConsistent(x) -> x to true
```
Comment (devcrocod) [inline multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt:40]: Why is the condition `x.dim.d == 2` needed here?

File: multik-cuda/src/test/kotlin/org/jetbrains/kotlinx/multik/cuda/math/CudaMathTest.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import org.jetbrains.kotlinx.multik.api.ndarray
+import org.jetbrains.kotlinx.multik.cuda.CudaMath
+import org.jetbrains.kotlinx.multik.ndarray.data.D2
+import org.jetbrains.kotlinx.multik.ndarray.data.NDArray
+
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class CudaMathTest {
```
Comment (devcrocod) [inline multik-cuda/src/test/kotlin/org/jetbrains/kotlinx/multik/cuda/math/CudaMathTest.kt:13]: empty

File: multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt (hunk 2 above)
Comment (Idutvuker) [reply multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt]: Fixed in 419f068

File: multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt (hunk 3 above)
Comment (Idutvuker) [reply multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt]: Added doc comment in 419f068

File: multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt (hunk 4 above)
Comment (Idutvuker) [reply multik-cuda/src/main/kotlin/org/jetbrains/kotlinx/multik/cuda/CudaLinAlg.kt]: If MultiArray is a vector (x.dim.d != 2) then the following condition is always false

File: multik-cuda/src/test/kotlin/org/jetbrains/kotlinx/multik/cuda/math/CudaMathTest.kt (hunk 5 above)
Comment (Idutvuker) [reply multik-cuda/src/test/kotlin/org/jetbrains/kotlinx/multik/cuda/math/CudaMathTest.kt]: Deleted file in 419f068

**Review by devcrocod — APPROVED** [review]

**Pull request merged**